safe/common/parameters/default_select_parameter_widget.py

```diff
@@ -151,3 +152,29 @@ def toggle_custom_value(self):
             self.custom_value.setDisabled(False)
         else:
             self.custom_value.setDisabled(True)
+
+    def toggle_input(self):
+        """Change behaviour of radio button based on input."""
+        # If input is a field, then disable all radio button.
+        if self.input.currentText() != 'No Field':
```

Comment:
Btw, `No field` is very specific. You can't put that in the default_select_parameter_widget. May be, you should add something like, a special attribute to the parameter. This attribute will store the string that will triger the behaviour. Then, you can set the string in the creation.

example:
```
if self.input.currentText() != self.special_content:
```